### DISCUSSION TITLE
Opentelemetry (collector) in the Android space - experience and best practice

### DISCUSSION THREAD

**Discussion by cforce** [open-telemetry/opentelemetry-collector-contrib#32022] [General]
I am currently starting to try out running Otel collection using the collector in android OS and would be keen on getting insights about specific setup and experience in this field. How is even Google is up to it?  I hear a lot of Otel and Google Cloud integration at [conferences](https://www.youtube.com/playlist?list=PLxNYxgaZ8RsdPmqSf-2KgznOmRP8lxdDL) but nothing in context of Android. What i do not see is whether OpenTelemetry plays a role in the future of Android at all (if not, why, if yes, how?)
While there is sdk wise integration via  https://github.com/open-telemetry/opentelemetry-android possible, what is the proposes setup for a gateway at edge or collector for a android os host metric and logs collection. There are a few documented APIs, e.g. NetworkStatsManager, App usage and similar APIs which might provide a good api to integrate in the collector.

**Comment by cforce** [comment]
Hello @braydonk, I'd love to hear your thoughts or ideas on this topic!

**Reply by braydonk** [reply]
Unfortunately I don't really have any suggestions here, I don't know anything about Android and can't make any recommendations. Hopefully someone else will be able to chime in!

**Reply by cforce** [reply]
Is there any update from Google’s side? I still don’t understand why Google/Android/AOSP doesn’t support using an OpenTelemetry Collector as a relay or system component. That approach would be much more efficient than having each app or system component send telemetry data individually.

Additionally, RUM (Real User Monitoring) data should ideally be sent through OpenTelemetry and could still be forwarded to Google Analytics if needed. The OpenTelemetry Android RUM implementation already provides far better insights than what’s available through basic analytics.
